src/tzdata/compile.jl

```diff
@@ -69,55 +69,120 @@ const DAYS = Dict(
     "Mon" => 1, "Tue" => 2, "Wed" => 3, "Thu" => 4, "Fri" => 5, "Sat" => 6, "Sun" => 7,
 )
 
+const LAST_DAY_OF_WEEK = Dict{String, Function}()
+
 # Create adjuster functions such as "lastSun".
 for (abbr, dayofweek) in DAYS
-    sym = Symbol("last" * abbr)
-    @eval (
+    str = "last" * abbr
+    sym = Symbol(str)
+    LAST_DAY_OF_WEEK[str] = @eval begin
         function $sym(dt)
             return dayofweek(dt) == $dayofweek &&
             dayofweekofmonth(dt) == daysofweekinmonth(dt)
         end
-    )
+    end
+end
+
+# Generate various DateFormats based the number of periods provided.
+const UNTIL_FORMATS = let
+    parts = split("yyyy uuu dd HH:MM:SS", ' ')
+    date_formats = Vector{DateFormat}(undef, length(parts))
```

Comment:
```julia
date_formats = map(i->DateFormat(join(parts[1:i], ' ')), 1:length(parts))
```
?

Reply:
```julia
date_formats = map(i->DateFormat(join(parts[1:i], ' ')), 1:length(parts))
```
```julia
date_formats = map(1:length(parts)) do i
    DateFormat(join(parts[1:i], ' '))
end
```
```julia
date_formats = [DateFormat(join(parts[1:i], ' ')) for i in 1:length(parts)]
```

Reply:
Or just remove the `date_formats` variable since it's returned and never used again.

Reply:
I tend to be biased against `map` for some reason. I do like it better in this case if I used `eachindex`

Reply:
```julia
const UNTIL_FORMATS = let parts = split("yyyy uuu dd HH:MM:SS", ' ')
    map(i->DateFormat(join(parts[1:i], ' ')), 1:length(parts))
end
```

Reply:
> biased against `map`

:scream: :sob:

Reply:
I find inline anonymous function syntax hard to read when there are a few pairs of parentheses going around

Reply:
What are your feelings on:
```julia
const UNTIL_FORMATS = let
    parts = split("yyyy uuu dd HH:MM:SS", ' ')
    map(i -> DateFormat(join(parts[1:i], ' ')), eachindex(parts))
end
```